envi/archs/arm/disasm.py

```diff
@@ -1906,6 +1909,19 @@ def involvesPC(self):
     def isDeref(self):
         return True
 
+    def setOperValue(self, op, emu=None, val=None):
+        # can't survive without an emulator
+        if emu == None:
+            return None
+
+        pubwl = self.pubwl >> 2
+        b = pubwl & 1
+
+        addr = self.getOperAddr(op, emu)
+
+        fmt = ("<L", "B")[b]
```

Comment:
thanks.  will do.  also should check PE(and others):
PE/__init__.py:674:        fmt = "<L"

envi/archs/i386/emu.py:933:        neweax = self.readMemoryFormat(esi, "<L")[0]
envi/archs/i386/emu.py:1484:        memval = struct.unpack("<L",self.readMemory(base+edi, 4))[0]
envi/archs/i386/emu.py:1508:        self.writeMemory(base+edi, struct.pack("<L", eax))
envi/cli.py:603:            pattern = struct.pack('<L', sval) # FIXME 64bit (and alt arch)
vtrace/archs/i386.py:142:                ebp,eip = struct.unpack("<LL",buf)
vtrace/platforms/solaris.py:105:        self.ctl.write(struct.pack("<L", PRSTOP))
vtrace/platforms/solaris.py:111:        self.writeCtl(struct.pack("<LL", PCRUN, 0))
vtrace/platforms/solaris.py:117:        self.writeCtl(struct.pack("<L", PCWSTOP))
vtrace/platforms/solaris.py:150:        priovec = struct.pack("<LLLL", PCWRITE, baddr, blen, addr)
vtrace/platforms/solaris.py:158:            addr,size = struct.unpack("<LL", mapdata[:8])
vtrace/platforms/solaris.py:159:            perms, = struct.unpack("<L", mapdata[80:84])


i'll leave these here, since i don't know the context.  but i'm guessing they all want to be <I